src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -4,27 +4,129 @@
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Constants.constElevator;
+import frc.robot.Robot;
 import frc.robot.RobotMap.mapElevator;
 
 public class Elevator extends SubsystemBase {
-  TalonFX elevatorLeftMotor;
-  TalonFX elevatorRightMotor;
+  TalonFX leftMotorFollower;
+  TalonFX rightMotorLeader;
   TalonFX elevatorLeftPivotMotor;
   TalonFX elevatorRightPivotMotor;
 
+  private Angle lastDesiredAngle = Degrees.zero();
+
+  Distance currentLeftPosition = Units.Inches.of(0);
+  Distance currentRightPosition = Units.Inches.of(0);
+  private Distance lastDesiredPosition;
+  MotionMagicExpoVoltage positionRequest = new MotionMagicExpoVoltage(0);
+
   /** Creates a new Elevator. */
   public Elevator() {
 
-    elevatorLeftMotor = new TalonFX(mapElevator.ELEVATOR_LEFT_CAN); // Elevator left motor
-    elevatorRightMotor = new TalonFX(mapElevator.ELEVATOR_RIGHT_CAN); // Elevator right motor
+    leftMotorFollower = new TalonFX(mapElevator.ELEVATOR_LEFT_CAN); // Elevator left motor
+    rightMotorLeader = new TalonFX(mapElevator.ELEVATOR_RIGHT_CAN); // Elevator right motor
     elevatorLeftPivotMotor = new TalonFX(mapElevator.ELEVATOR_LEFT_PIVOT_CAN); // Elevator left pivot motor
     elevatorRightPivotMotor = new TalonFX(mapElevator.ELEVATOR_RIGHT_PIVOT_CAN); // Elevator right pivot motor
 
+    lastDesiredPosition = Units.Inches.of(0);
     // Set default motor configurations if needed
     // e.g., elevatorLeftMotor.configFactoryDefault();
+    leftMotorFollower.getConfigurator().apply(constElevator.ELEVATOR_MOTOR_CONFIG);
+    rightMotorLeader.getConfigurator().apply(constElevator.ELEVATOR_MOTOR_CONFIG);
+    elevatorLeftPivotMotor.getConfigurator().apply(constElevator.ELEVATOR_PIVOT_CONFIG);
+    elevatorRightPivotMotor.getConfigurator().apply(constElevator.ELEVATOR_PIVOT_CONFIG);
+  }
+
+  public AngularVelocity getRotorVelocity() {
+    return rightMotorLeader.getRotorVelocity().getValue();
+  }
+
+  public boolean isRotorVelocityZero() {
+    return getRotorVelocity().isNear(Units.RotationsPerSecond.zero(), 0.01);
+  }
+
+  public Distance getLastDesiredPosition() {
+    return lastDesiredPosition;
+  }
+
+  public Distance getElevatorPosition() {
+    if (Robot.isSimulation()) {
+      return getLastDesiredPosition();
+    }
+    return Units.Inches.of(rightMotorLeader.getPosition().getValueAsDouble());
+  }
+
+  public double getRightPosition(Distance height) {
+    return rightMotorLeader.getPosition().getValueAsDouble();
+  }
+
+  public double getLeftPosition(Distance height) {
+    return leftMotorFollower.getPosition().getValueAsDouble();
+  }
+
+  public void setElevatorPosition(Distance height) {
+    rightMotorLeader.setControl(positionRequest.withPosition(height.in(Units.Inches)));
+    leftMotorFollower.setControl(new Follower(rightMotorLeader.getDeviceID(), true));
+    lastDesiredPosition = height;
+  }
+
+  public boolean isAtSpecificSetpoint(Distance setpoint) {
```

Comment:
Since we have both a lift and pivot we need to make sure it is clear which we are talkinkg about
```suggestion
  public boolean isLiftAtSpecificSetpoint(Distance setpoint) {
```